api/app/services/availability_service.py

```diff
@@ -76,6 +76,9 @@ def get_available_slots(office: Office, days: [datetime], format_time: bool = Tr
                             end_time = add_delta_to_time(end_time, minutes=appointment_duration,
                                                          timezone=office.timezone.timezone_name)
 
+                # Sort the slot by time for the day
```

Comment:
This won't hurt, but I think the query is getting all appointments by time and the time slot calculation starts with ascending order of slots.